crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -1,4 +1,120 @@
-//! Types for creating and storing [`Observer`]s
+//! Observers are a push-based tool for responding to [`Event`]s.
+//!
+//! ## Observer targeting
+//!
+//! Observers can be "global", listening for events that are both targeted at and not targeted at any specific entity,
+//! or they can be "entity-specific", listening for events that are targeted at specific entities.
+//!
+//! They can also be further refined by listening to events targeted at specific components
+//! (instead of using a generic event type), as is done with the [`OnAdd`] family of lifecycle events.
+//!
+//! When entities are observed, they will receive an [`ObservedBy`] component,
+//! which will be updated to track the observers that are currently observing them.
+//!
+//! Currently, [observers cannot be retargeted after spawning](https://github.com/bevyengine/bevy/issues/19587):
+//! despawn and respawn an observer as a workaround.
+//!
+//! ## Writing observers
+//!
+//! Observers are systems that implement [`IntoObserverSystem`] and listen for [`Event`]s that match their
+//! type and target(s).
```

Comment:
```suggestion
//! Observers are systems that implement [`IntoObserverSystem`] and listen for [`Event`]s matching their
//! type and target(s).
```
Might be better to remove the second "that"?